subsys/bluetooth/services/fast_pair/crypto/fp_crypto_common.c

```diff
@@ -60,3 +71,126 @@ int fp_crypto_account_key_filter(uint8_t *out, const struct fp_account_key *acco
 	}
 	return 0;
 }
+
+int fp_crypto_aes_ctr_encrypt(uint8_t *out, const uint8_t *in, size_t data_len, const uint8_t *k,
+			      const uint8_t *nonce)
+{
+	size_t block_cnt;
+	uint8_t aes_input[FP_CRYPTO_AES128_BLOCK_LEN];
+	uint8_t aes_block[FP_CRYPTO_AES128_BLOCK_LEN];
+	int err;
+
+	if (data_len % FP_CRYPTO_AES128_BLOCK_LEN == 0) {
+		block_cnt = data_len / FP_CRYPTO_AES128_BLOCK_LEN;
+	} else {
+		block_cnt = (data_len / FP_CRYPTO_AES128_BLOCK_LEN) + 1;
+	}
+
+	memset(&aes_input[AES_INPUT_PAD_START], 0, AES_INPUT_PAD_LEN);
+	memcpy(&aes_input[AES_INPUT_NONCE_START], nonce, FP_CRYPTO_NONCE_LEN);
+
+	memcpy(out, in, data_len);
+	for (size_t i = 0; i < block_cnt; i++) {
```

Comment:
What is a maximum value of `block_cnt`? Is it safe to cast it to `uint8_t` ?

Reply:
It's not specified. Fast Pair specification (see image below) casts it to `uint8_t`. I believe it shouldn't be a problem even if `block_cnt` exceeded `uint8_t` range which btw. is very unlikely. In current specification we use that function only to encrypt personalized name of the device so the name would have to exceed 256*16 characters.
![image](https://user-images.githubusercontent.com/86837740/177812052-7ddc15ee-32f8-4d9c-9bce-48bba979300f.png)